### PR TITLE
Add contact sensor on HomeKit & some fix

### PR DESCRIPTION
**Pull request by bertrandda**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.
- [x] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [x] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Add contact sensors (door, window...)
Fix bulb brightness range
Notify HomeKit of new value

### CODE REVIEW

File: server/services/homekit/lib/notifyChange.js
Comment (Pierre-Gilles) [inline server/services/homekit/lib/notifyChange.js:12]: Just a small feedback: Sending every device state change to HomeKit is super heavy, some device send many state changes per second, and usually the only relevant information is the last value.

In the Alexa/Google Home integration, we send only 1 state per 5 seconds maximum per device. That's what Google recommended. Only the "last" value is relevant. 

See: https://github.com/GladysAssistant/Gladys/blob/master/server/lib/gateway/gateway.forwardDeviceStateToAlexa.js#L109

and: https://github.com/GladysAssistant/Gladys/blob/master/server/lib/gateway/index.js#L55

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Nice PR! 🙏

File: server/services/homekit/lib/notifyChange.js
Comment (bertrandda) [reply server/services/homekit/lib/notifyChange.js:12]: I improved that as asked (see https://community.gladysassistant.com/t/integration-homekit-et-siri/7493/48)

Comment (Pierre-Gilles) [reply server/services/homekit/lib/notifyChange.js:12]: Looks way better now! Thanks for the changes

**Review by Pierre-Gilles — DISMISSED** [review]
Thanks for your changes 🙏

Good for me ! 

Just need confirmation before merging, this can be deployed in PROD without any issues? People won't have to repair anything?

File: server/services/homekit/lib/createBridge.js
Comment (Pierre-Gilles) [inline server/services/homekit/lib/createBridge.js]: I just re-read the code and found this, it seems to me that every time a home kit bridge is created, it creates a subscription on `this.gladys.event.on(EVENTS.TRIGGERS.CHECK`. It'll create multiple subscription, no ?

Comment (bertrandda) [reply server/services/homekit/lib/createBridge.js]: Indeed, it will create multiple subscriptions 🤔 for example when we reload the bridge after adding new accessory to Gladys. Thanks to delay before notify it should send only 1 message but need to improve that.
I quickly check `lib/event` file not sure how to delete previous one before create the new one. Any idea?

Comment (bertrandda) [reply server/services/homekit/lib/createBridge.js]: Maybe we need to implement `removeListener()` in `Event`?

Comment (Pierre-Gilles) [reply server/services/homekit/lib/createBridge.js]: Indeed, we miss a removeListener!

Comment (Pierre-Gilles) [reply server/services/homekit/lib/createBridge.js]: https://nodejs.org/api/events.html#emitterremovelistenereventname-listener

Comment (bertrandda) [reply server/services/homekit/lib/createBridge.js]: Done! I added `removeListener` on our Event implementation and use it for HomeKit reload

File: server/test/lib/event/event.test.js
Comment (Pierre-Gilles) [inline server/test/lib/event/event.test.js:19]: I don't think this is testing the behavior, because our event emitting system is async, so even if you remove the 

> eventEmitter.removeListener('test', cb);

Line, you'll see that alreadySeen will still be false

Comment (bertrandda) [reply server/test/lib/event/event.test.js:19]: Yes I was not sure about this test. I just found a better way by implementing and using `listenerCount`

**Review by Pierre-Gilles — DISMISSED** [review]
Looks good to me now! 

Can it be merged to PROD? Was it tested in real life with latest changes? Will users needs to re-pair their devices?

**Comment by bertrandda** [comment]
> Can it be merged to PROD? Was it tested in real life with latest changes? Will users needs to re-pair their devices?

I am going to create a  last image with last master changes and ask for a last test but yes I think everything fonctionnally as been tested.
Users will not have to repair their devices, just define new device's room in Home app

**Comment by Pierre-Gilles** [comment]
> I am going to create a last image with last master changes and ask for a last test but yes I think everything fonctionnally as been tested.

Ok, let me know here when it's good for merge.

**Comment by Pierre-Gilles** [comment]
@bertrandda  Same comment here, if we don't have more testers in one week, and you tested carefully on your side, I'm ok to merge on master :) Ping me if I forget!

**Pull request merged**
